service/controller/app/v1/resource/tiller/create.go

```diff
@@ -22,7 +22,15 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 
 	if key.InCluster(cr) {
 		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("app %#q in %#q uses InCluster kubeconfig no need to install tiller", cr.Name, cr.Namespace))
-		r.logger.LogCtx(ctx, "level", "debug", "message", "cancelling the resource")
+		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
+		return nil
+	}
+
+	// Resource is used to bootstrap chart-operator in tenant clusters.
+	// So for other apps we can skip this step.
```

Comment:
This is the actual fix. If tiller is broken we only wait for the chart-operator app.